### PR TITLE
Set always_run: true for DV GC lane

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 1]
```diff
@@ -260,7 +260,6 @@ presubmits:
       preset-shared-images: "true"
     max_concurrency: 11
     name: pull-kubevirt-e2e-k8s-1.24-sig-storage-dv-gc
-    optional: true
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:263]: I would not yet make it required, it seems like the lane is not yet stable enough: https://prow.ci.kubevirt.io/job-history/gs/kubevirt-prow/pr-logs/directory/pull-kubevirt-e2e-k8s-1.24-sig-storage-dv-gc

Please do not forget that a failing required lane will block merging a PR.

**Review by dhiller — CHANGES_REQUESTED** [review]

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 1 above)
Comment (arnongilboa) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:263]: No problem, I'll battle the lane flakiness in https://github.com/kubevirt/kubevirt/pull/8351

**Review by dhiller — APPROVED** [review]
/approve

**Comment by fabiand** [comment]
@dhiller did we consider the impact on our ci resources?
Also what was the reasoning for enabling it always?

IOW it's a big change for CI and there is no discussion at all ...

**Comment by arnongilboa** [comment]
> @dhiller did we consider the impact on our ci resources? Also what was the reasoning for enabling it always?
> 
> IOW it's a big change for CI and there is no discussion at all ...

Having GC enabled affects the behavior of all created DVs, so we must have at least one lane with GC.

In the near future, GC will be enabled by default, so this specific lane may get replaced by a single lane with no-GC.

**Pull request merged**